packages/store-indexer/src/sqlite/formatSqlQuery.ts

```diff
@@ -0,0 +1,50 @@
+import sqlParser from "node-sql-parser";
+
+type AstNode = {
+  type?: string;
+  [key: string]: unknown;
+};
+
+/**
+ * Transforms camelCase identifiers to snake_case in SQL queries
+ *
+ * @param sqlQuery The SQL query to transform
+ * @returns The transformed SQL query
+ */
+export function formatSqlQuery(sqlQuery: string): string {
+  const parser = new sqlParser.Parser();
+  const ast = parser.astify(sqlQuery);
+
+  updateNode(ast);
+
+  return parser.sqlify(ast, { database: "sqlite" });
+}
+
+function updateNode(node: unknown): void {
+  if (Array.isArray(node)) {
+    node.forEach((item) => updateNode(item));
+    return;
+  }
+
+  if (node && typeof node === "object") {
+    const astNode = node as AstNode;
+
+    if (
+      astNode.type === "column_ref" &&
+      typeof astNode.column === "string" &&
+      /^[a-z]+[A-Z][a-z]*$/.test(astNode.column)
+    ) {
+      astNode.column = camelToSnakeCase(astNode.column);
```

Comment:
thoughts on using `change-case` lib for this instead of our own implementation? we already use this in store-sync for sqlite table names

Reply:
also don't think it's necessary to test if the column has something that looks like camel case, we can just change it all to snake case if that's the expected format